app/src/main/java/com/hamdan/forzenbook/data/repository/LoginRepositoryImpl.kt

```diff
@@ -7,13 +7,15 @@ class LoginRepositoryImpl(
     private val dao: LoginDao,
     private val service: LoginService
 ) : LoginRepository {
-    override suspend fun getToken(email: String): LoginData? {
-        //check database for token
-        //if no token in data base get from service
+    override suspend fun requestValidation(email: String): Int {
         return try {
-            LoginData(service.getToken().body()?.token)
+            service.requestValidation(email).code()
         } catch (e: Exception) {
-            null
+            0
```

Comment:
No returning an int

Reply:
It doesn't make sense to return an int, especially http status codes. No one below this layer cares about codes

Reply:
changed to be an enum